functions/New-DbaDbUser.ps1

```diff
@@ -0,0 +1,247 @@
+                        $Name = $Username;
+                    }
+                } #switch
+                
+                # Does user exist with same name?
+                if ($existingUser = $db.Users[$Name]) {
+                    if ($force) {
+                        if ($Pscmdlet.ShouldProcess($existingUser, "Dropping existing user $Name because -Force was used")) {
+                            try {
+                                $existingUser.Drop()
+                            }
+                            catch {
+                                Stop-Function -Message "Could not remove existing user $Name, skipping." -Target $Name -Continue
+                                return
+                            }
+                        }
+                    }
+                    else {
+                        Stop-Function -Message "User $Name already exists and -Force was not specified" -Target $Name
+                        return
+                    }
+                }
+                
+                
+                if ($Pscmdlet.ShouldProcess($db, "Creating user $Name")) {
+                    try {
+                        $smoUser = New-Object Microsoft.SqlServer.Management.Smo.User;
+                        $smoUser.Parent = $db;
+                        $smoUser.Name = $Name;
+                        
+                        If ($PSBoundParameters.Keys -contains 'Login' -and $Login.GetType().Name -eq 'Login') { $smoUser.Login = Login }
+                        $smoUser.UserType = $UserType;
+                        
+                        $smoUser.Create();
+                        
+                    }
+                    catch {
+                        Stop-Function -Message "Failed to add user $Name in $db to $instance" -Category InvalidOperation -ErrorRecord $_ -Target $instance -Continue;
+                        return
+                    }
+                    $smoUser.Refresh();
+                    
+                    if ($PSBoundParameters.Keys -contains 'Username' -and $smoUser.Name -ne $Username) {
+                        $smoUser.Rename($Username);
+                    }
+                    
+                    # if ( $PSBoundParameters.Keys -contains 'Password' ) {
+                    #     $smoUser.ChangePassword('',$Password);
+                    #     $smoUser.Alter();
+                    #     $smoUser.Refresh();
+                    # }
+                    
+                    Write-Message -Level Verbose -Message "Successfully added $smoUser in $db to $instance."
+                    
+                }
+                
+                #Display Results
+                Get-DbaDatabaseUser -SqlInstance $server.Name -Database $db.Name | Where-Object name -eq $smoUser.Name
+            } #foreach ($db in $databases)
+        } #foreach ($instance in $SqlInstance)
+    }
+    end {
+        if (Test-FunctionInterrupt) { return }
```

Comment:
All the returns in this command need to be removed if you are going to allow processing of multiple databases, logins, and users. Once that is done this end block can be removed as well.